### ISSUE TITLE
Collaborating the credential-helper with aws SSO fails

### ISSUE THREAD

**Issue by JoshuaGhost** [awslabs/amazon-ecr-credential-helper#573]
I have been trying to collaborate the ecr + credential-helper and aws SSO authentication. Here is what i did and how it fails:
First, i set up the sso-authentication with
```
aws configure sso-session
```
and 
```
aws configure sso
```
the automatically generated ~/.aws/config looks like this:
```
[sso-session my-sso]
sso_start_url = https://***
sso_region = ***
sso_registration_scopes = sso:account:access
[profile p1]
sso_session = my-sso
sso_account_id = ***
sso_role_name = ***
region = ***
output = json
```
I can now login using 
```
aws sso login --profile p1
```

Then i installed amazon-ecr-credential-helper under Ubuntu 20.04.6 LTS through go install. I also updated the ~/.docker/config.json according to the readme.md as follows:

```
{
  "credHelpers": {
    /*helper settings for other IaaS providers like GCP*/
    "<aws-user-id>.dkr.ecr.<aws-cli-region>.amazonaws.com/<ecr-registry-name>": "ecr-login"
  },
    "credsStore": "ecr-login"
}
```

But now after logging in using the sso, i can't push nor pull the image to/from the ECR. The error message says that i have "**no basic auth credentials**"

I could acually use the trick suggested [here](https://docs.aws.amazon.com/AmazonECR/latest/userguide/docker-push-ecr-image.html) to get the password and re-direct it to the `--password-stdin` argument of `docker login`. But this command has the following side-effects:

1. It removes my ~/.docker/config.json file and create a new one with only the auth passwords.
2. It stores the password in ~/.docker/config.json file in (base64-encoded) plain text

I don't like either of these side-effects so could you please help me by telling me the most practical way of using the tool chain?

After searching on the internet and within the issues of this repo, i found [this](https://github.com/awslabs/amazon-ecr-credential-helper/issues/522) issue mentioned an influent ochestration of sso and credential helper. @austinvazquez suggested to update the credential-helper. But for me updating this doesn't resolve the problem.

**Comment by gopipal** [comment]
You may need to `docker logout "<aws-user-id>.dkr.ecr.<aws-cli-region>.amazonaws.com"` since docker can cache old creds. 

I am using the latest aws credentials helper and it works with aws sso
```bash
go install github.com/awslabs/amazon-ecr-credential-helper/ecr-login/cli/docker-credential-ecr-login@latest
sudo cp $GOPATH/bin /usr/bin/.
```
This is my `~/.docker/config.json`
```json
{
	"credHelpers": {
		"<aws-user-id>.dkr.ecr.<aws-cli-region>.amazonaws.com": "ecr-login"
	}
}
```

**Comment by blackliner** [comment]
I can agree the latest version works, 0.3.1 that I had from Ubuntu 20.04 was failing.

**Comment by austinvazquez** [comment]
Agreed @blackliner, some distributions are having older credential helper versions.

For others, the AWS SSO feature was added in v0.6.0. So just be sure to double check the versions. If your distribution does not have newer versions available, perhaps check if any of the statically linked binaries published in [releases](https://github.com/awslabs/amazon-ecr-credential-helper/releases) works for you.

For the `not implemented` issue, an enhancement was added in v0.9.0 which allowed users to specify `AWS_ECR_IGNORE_CREDS_STORAGE=true` environment variable. The reason for the error is the amazon ecr credential helper does not store any credentials, so whenever users perform a `docker login` then docker calls the credential helpers `Add` function. With credential helpers, users should not have to do a login command if their Docker configuration is setup correctly; however, I have seen some clients always do a login command. For these use cases, the environment variable was added to silently ignore calls to `Add`.

Closing this issue as complete as I believe both issues have already been addressed. @JoshuaGhost feel free to reopen if you have any more questions.

**Comment by WellspringCS** [comment]
I am wondering if my issue is the same as this one. If in WSL we use the credential helper and the profile uses sso_session, this error is thrown when docker attempts to pull an ECR image:
panic: profile "<your-profile>" is configured to use SSO but is missing required configuration: sso_region, sso_start_url.

Then after some more output, the final error message is: no basic auth credentials.

If we remove the dependency on sso_session from the profile and instead include sso_region, sso_start_url directly in the profile, there is no error and the image is pulled.

We have replicated this problem on several different machines.
